### PR TITLE
CFE-2596: Keys in the lock database, cf_lock.lmdb, are now human-readable

### PR DESCRIPTION
**Pull request by olehermanse**
LMDB keys are limited to 511 bytes, by default. To get around
this, the keys were changed to be hashes some time ago.
This change effectively reverts that commit, making the keys
normal strings by default, and overwriting the last part with a
hash, if necessary (if the key would exceed 511 bytes).

### CODE REVIEW

**Comment by olehermanse** [comment]
@cf-bottom jenkins, please.

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/2900/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/2900/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-2900/

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/2916/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/2916/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-2916/

File: libpromises/locks.c [hunk 1]
```diff
@@ -177,29 +180,12 @@ static inline void log_lock(const char *op,
...
-        ohash[i*2]   = lookup[digest[i] >> 4];
-        ohash[i*2+1] = lookup[digest[i] & 0xf];
-    }
-    ohash[16*2] = '\0';
-
-    if (!strncmp(istring, "lock.track_license_bundle.track_license", 39))
-    {
-        ohash[0] = 'X';
-    }
+    assert(LMDB_MAX_KEY_SIZE == 511);
```
Comment (vpodzime) [inline libpromises/locks.c]: This *assert* doesn't make much sense here. The function should work with any value for `LMDB_MAX_KEY_SIZE` that passes the following asserts, right?

Comment (olehermanse) [reply libpromises/locks.c]: It's just an attempt to get people to review this code if they ever change the constant.

**Comment by olehermanse** [comment]
@cf-bottom Jenkins again, please.

File: libpromises/locks.c (hunk 1 above)
Comment (olehermanse) [reply libpromises/locks.c]: Removed.

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/2948/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/2948/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-2948/

**Pull request merged**